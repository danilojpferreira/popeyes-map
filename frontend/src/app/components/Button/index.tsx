import React, { CSSProperties, memo, MouseEventHandler, ReactNode } from 'react'
import './styles.scss'

interface Props {
  disabled?: boolean
  onClick?: MouseEventHandler<HTMLButtonElement>
  text: string
  className?: string
  secondary?: boolean
  style?: CSSProperties
  rightIcon?: ReactNode
}

const Button = ({
  disabled,
  onClick,
  text,
  className = '',
  style,
  secondary,
  rightIcon,
}: Props): JSX.Element => (
  <button
    type="button"
    disabled={disabled}
    onClick={onClick}
    className={`button ${className} ${secondary && 'button-secondary'}`}
    style={{
      ...style,
    }}
  >
    {text}
    {rightIcon ? <div className="button_right-icon">{rightIcon}</div> : null}
  </button>
)

export default memo(Button)
