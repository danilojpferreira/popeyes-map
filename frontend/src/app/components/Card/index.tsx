import React, { memo, ReactNode } from 'react'
import './styles.scss'

interface Props {
  children: ReactNode
  className?: string
}

const Card = ({ children, className = '' }: Props): JSX.Element => (
  <div className={`card ${className}`}>{children}</div>
)

export default memo(Card)
