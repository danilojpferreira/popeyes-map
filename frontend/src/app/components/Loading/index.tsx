import React, { memo } from 'react'
import { useSelector } from 'react-redux'
import { CircularProgress } from '@material-ui/core'
import { Card } from '..'
import { RootState } from '../../store/reducers'
import './styles.scss'

const Loading = (): JSX.Element => {
  const { isLoading } = useSelector(
    (state: RootState) => state.configurationsState
  )
  return (
    isLoading && (
      <div className="loading">
        <Card className="loading_card">
          <CircularProgress />
          <p>Please wait, loading...</p>
        </Card>
      </div>
    )
  )
}

export default memo(Loading)
