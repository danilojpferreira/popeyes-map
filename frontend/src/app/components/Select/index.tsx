/* eslint-disable @typescript-eslint/ban-ts-comment */
import React, { memo } from 'react'
import {
  FormControl,
  InputLabel,
  MenuItem,
  Select as InputSelect,
} from '@material-ui/core'
import './styles.scss'

interface Props {
  id?: string
  label: string
  selected?: string | number
  options: {
    value: string | number | boolean | null
    label: string
  }[]
  onSelect?: (value: number | string) => void
  title?: string
  className?: string
  disabled?: boolean
}

const Select = ({
  id,
  label,
  selected,
  options,
  onSelect,
  title,
  className,
  disabled,
}: Props): JSX.Element => (
  <div className={`select ${className}`}>
    {title && <p className="select_title">{title}</p>}
    <FormControl
      variant="filled"
      className="select_form-control"
      disabled={disabled}
    >
      <InputLabel id={id}>{label}</InputLabel>
      <InputSelect
        labelId={id}
        value={selected}
        // @ts-ignore
        onChange={({ target }) => onSelect(target.value)}
      >
        {options.map(item => (
          // @ts-ignore
          <MenuItem key={Math.random()} value={item.value}>
            {item.label}
          </MenuItem>
        ))}
      </InputSelect>
    </FormControl>
  </div>
)

export default memo(Select)
