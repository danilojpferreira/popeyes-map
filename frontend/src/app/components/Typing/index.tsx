/* eslint-disable no-await-in-loop */
/* eslint-disable no-restricted-syntax */
import React, { memo, useEffect, useState } from 'react'
import { sleep } from '../../utils/utils'
import './styles.scss'

interface Props {
  sentences: string[]
  onFinish: () => void
}

const Typing = ({ sentences = [], onFinish }: Props): JSX.Element => {
  const [text, setText] = useState('')

  useEffect(() => {
    const run = async () => {
      for (const sentence of sentences) {
        for (const i of sentence.split('')) {
          // eslint-disable-next-line no-loop-func
          setText(prev =>
            prev.length === sentences.join().length + (sentences.length - 1)
              ? prev
              : `${prev}${i}`
          )
          await sleep(50)
        }
        setText(prev =>
          prev.length === sentences.join().length + (sentences.length - 1)
            ? prev
            : `${prev} `
        )
        await sleep(700)
      }
    }
    run()
  }, [sentences])

  useEffect(() => {
    if (text.length === sentences.join().length + sentences.length - 1)
      onFinish()
  }, [onFinish, sentences, text.length])

  return <p className="typing">{text}</p>
}

export default memo(Typing)
