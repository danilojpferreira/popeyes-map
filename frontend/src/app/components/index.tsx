export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as Loading } from './Loading'
export { default as Typing } from './Typing'
export { default as Select } from './Select'
