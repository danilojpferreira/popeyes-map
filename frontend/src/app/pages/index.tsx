export { default as Main } from './Main'
export { default as Maps } from './Maps'
export { default as Error } from './Error'
