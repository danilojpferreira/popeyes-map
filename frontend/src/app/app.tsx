import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { ToastContainer } from 'react-toastify'
import { injectStyle } from 'react-toastify/dist/inject-style'
import { store } from './store'
import { Loading } from './components'
import { Error, Main, Maps } from './pages'

injectStyle()

const App = (): JSX.Element => (
  <BrowserRouter>
    <Provider store={store}>
      <Loading />
      <Routes>
        <Route path="/" element={<Main />} />
        <Route path="/maps/:map/:vehicle/:reverse" element={<Maps />} />
        <Route path="/maps/:map/:vehicle" element={<Maps />} />
        <Route path="/404" element={<Error />} />
      </Routes>
      <ToastContainer autoClose={3000} />
    </Provider>
  </BrowserRouter>
)

export default App
