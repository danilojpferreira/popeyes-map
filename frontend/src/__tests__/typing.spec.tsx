/* eslint-disable no-alert */
/* eslint-disable @typescript-eslint/ban-ts-comment */
import renderer from 'react-test-renderer'
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Typing } from '../app/components'

Enzyme.configure({ adapter: new Adapter() })

const sentences = ['']

const base = <Typing onFinish={() => null} sentences={sentences} />

describe('<Typing />', () => {
  // @ts-ignore
  it('Should render correctly', async () => {
    const component = renderer.create(base)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render a label omponent', () => {
    const component = shallow(base)
    expect(component.find('.typing').text()).toEqual('')
  })
})
