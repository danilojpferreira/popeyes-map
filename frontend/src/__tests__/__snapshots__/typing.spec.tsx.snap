// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typing /> Should render correctly 1`] = `
<p
  className="typing"
>
  
</p>
`;
